Google summer of code (#239)
mautic/mautic-community-handbook: docs/contributing/google_summer_of_code.rst

.. image:: images/Summer_of_Code.png
   :width: 600

Google Summer of Code at Mautic
###############################

> This page contains information about Mautic's participation in Google Summer of Code, often abbreviated to GSoC. Please feel free to contact us via Slack below for clarifications and more information.

Reviewer: ```suggestion
This page contains information about Mautic's participation in Google Summer of Code, often abbreviated to GSoC. Please feel free to contact the team via Slack below for clarifications and more information.
```

> What can I do today?

Reviewer: ```suggestion
What to get started with today?
```

> ********************

Reviewer: Also fix the underline ;)

Today you should read through the ideas for projects, and start working on your application. Get familiar with Mautic, make some contributions so you're familiar with the workflow, and get to know the community. Join the :xref:`Mautic Community Slack` channel - #gsoc-discuss - and post any questions you have there, after reading this page fully and also reviewing the project proposals.

Student Information
*******************

Do you want to contribute to one of the world's first Open Source Marketing Automation platform? GSoC could be your chance. Please refer to the information below.

> If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling document which we maintain year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an “I'm interested” email or Slack message - please respect their time.

Reviewer: ```suggestion
If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling submission process which is maintained year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an “I'm interested” email or Slack message - please respect their time.
```

> If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling submission process which is maintained year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an “I'm interested” email or Slack message - please respect their time.

Reviewer: ```suggestion
If you're interested in working on a mentor-submitted idea, read the :xref:`GSOC project ideas` and about the project and the related technology. Please note this is a rolling submission process which Mautic maintains year-round. There is no need to contact the mentor unless you have a specific question about the project. Don't send an ``“I'm interested”`` email or Slack message - please respect their time.
```

Vale fix

People who contribute in our community and do good work are noticed. They get hired for jobs from it. Experience and reputation in the Mautic Project communities can be influential in your career in many ways.

You should know that contributing to FOSS doesn't require you to have super programming skills, or super-anything else. You just need be interested, curious, and willing to become comfortable being productively lost. This is the state of learning. You learn by finding your way around and figuring things out with the support of your mentor and the community.

Student Responsibilities
************************

You are the key ingredient for your project's success. This project is important to you, your mentor, and the entire Mautic Community.

Your responsibilities include:

- Communicating early and often with your mentor.
- Updating every week about what you're learning, how you're doing, challenges and successes. This is key way to keep the entire Mautic Community informed.

> - Working with your mentor on realistic achievable milestones that provide for regular deliverables and feedback.

Reviewer: ```suggestion
- Working with your mentor on realistic achievable milestones that provide for regular deliverable targets and feedback.
```

- Attending the brief student calls, as announced
- Being accountable for your success and your actions

Student Application
*******************

Please read and follow the student application process below.

Google Summer of Code Application Process
=========================================

> Interested in applying for Google Summer of Code with Mautic? Here is how you do that!

Reviewer: ```suggestion
Interested in applying for Google Summer of Code with Mautic? Here is how you do that.
```

These links may be helpful:

- :xref:`GSOC Official Site`

Communicate
~~~~~~~~~~~

Begin communicating with the community. Our community members can help you understand the existing project proposals. If you're going to try and propose your own project, this is a great way to understand the needs of the project and find a mentor.

- Join us on :xref:`Mautic Community Slack` at ``#gsoc-discuss``
- Join the Product Team chat in ``#t-product``
- Get involved with our weekly Open Source Friday sprints by testing bug fixes and new features

Don't forget to contact mentors on projects that interest you to learn more. If you're going to propose a project you must have a mentor at the time of proposal.

Apply
~~~~~

Student applications open on March 18 2024 - 18:00 UTC.

The application process consists of 3 steps:

1. Create an account on the :xref:`Mautic Community Forums`
2. Join the #gsoc-discuss and #t-product channels on Slack

> 3. Submit the application/proposal including all requirements in the next section at the Google Summer of Code Site. We **strongly** encourage you to not submit your final PDF until you get feedback about the proposal from your mentor(s).

Reviewer: ```suggestion
3. Submit the application/proposal including all requirements in the next section at the Google Summer of Code Site. We **strongly** encourage you to not submit your final PDF until you get feedback about the proposal from your mentors.
```

> Help is available! Share your application in progress and get feedback. The mentors can help verify your understanding of the project and provide guidance on whether your plan is good.

Reviewer: ```suggestion
Help is available. Share your application in progress and get feedback. The mentors can help verify your understanding of the project and provide guidance on whether your plan is good.
```

If you are submitting an original proposal you **must** find a mentor. Proposals without mentors will be rejected. Sharing your proposal is key to doing this.

If you don't want to share your proposal publicly consider using something like a Google Doc and sharing a link with the mentor/reviewers.

Google Summer of Code Application and Supplemental Material Requirements
------------------------------------------------------------------------

In addition to the information Google requests, the Mautic Project requires you to submit the following information:

- What project are you applying to? Seriously consider choosing a project for proposal from ideas list. While it is possible to propose your own project, it is often more successful for you to select one that is already proposed by a mentor.
- Contact Information - List your Name, Email Address, Blog URL, and any additional contact information you think we need. We tend to email so be sure you're reading your email.
- About You - Consider these questions

>   - Do you have any past involvement with the Mautic Project or another Open Source project as a contributor? What kind of contributions have you made (code and non-code)? Please share any public contributions you have made (urls, etc.)

Reviewer: ```suggestion
  - Do you have any past involvement with the Mautic Project or another Open Source project as a contributor? What kind of contributions have you made (code and non-code)? Please share any public contributions you have made (URLs, etc.)
```

  - What are you studying? Why?
  - What motivates you to do your best? How do you stay focused? Briefly describe how you organize your work.

- Why Us? - Consider these questions

  - Why do you want to work with the Mautic Project?

>   - Do you plan to continue contributing to the Mautic Project after GSoC? If yes, what project(s) are you interested with?

Reviewer: ```suggestion
  - Do you plan to continue contributing to the Mautic Project after GSoC? If yes, what projects are you interested with?
```

- Why This Project? - Consider these questions

  - Which specific project interests you? If you are proposing your own project, do it here. If you are proposing the project, explain why it benefits the community.

>   - Have you contacted the mentor(s) listed for the project?

Reviewer: ```suggestion
  - Have you contacted the mentors listed for the project?
```

  - What relevant experience do you have for this project?
  - What do you want to gain from this project?
  - Describe this project in your own words. What does it mean to you? What do you think the final deliverable is?

- Why should we choose you over other applicants?

> - Propose a rough schedule for this project. Ideally you will have regular deliverables throughout the entire period.

Reviewer: ```suggestion
- Propose a rough schedule for this project. Ideally you'll have regular deliverable tasks throughout the entire period.
```

> - What are you specific plans from May to September (eg: Exams in university, internships, vacations..etc). We need to know if there are periods of time where you are likely to be unavailable. We are better at accommodating these periods if we know of them in advance.

Reviewer: ```suggestion
- What are you specific plans from May to September (for example, exams in university, internships, vacations, etc). We need to know if there are periods of time where you are likely to be unavailable. We are better at accommodating these periods if we know of them in advance.
```
